include/pmacc/meta/ExecuteIf.hpp

```diff
@@ -0,0 +1,117 @@
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <utility>
+
+namespace pmacc
+{
+    namespace meta
+    {
+        namespace detail
+        {
+            /** Conditional functor execution
+             *
+             * @tparam T_condition condition for executing a functor
+             */
+            template<bool T_condition>
+            struct IfElse;
+
+            /** Specialization if condition is true
+             *
+             * Execute trueFunctor passed to operator()
+             */
+            template<>
+            struct IfElse<true>
+            {
+                template<typename T_ConditionTrueFunctor, typename T_ConditionFalseFunctor, typename... T_Args>
+                void operator()(T_ConditionTrueFunctor&& trueFunctor, T_ConditionFalseFunctor&&, T_Args&&... args)
+                    const
+                {
+                    trueFunctor(std::forward<T_Args>(args)...);
+                }
+            };
+
+            /** Specialization if condition is false
+             *
+             * Execute falseFunctor passed to operator()
+             */
+            template<>
+            struct IfElse<false>
+            {
+                template<typename T_ConditionTrueFunctor, typename T_ConditionFalseFunctor, typename... T_Args>
+                void operator()(T_ConditionTrueFunctor&&, T_ConditionFalseFunctor&& falseFunctor, T_Args&&... args)
+                    const
+                {
+                    falseFunctor(std::forward<T_Args>(args)...);
+                }
+            };
+        } // namespace detail
+
+        /** Conditional functor execution
+         *
+         * Evaluate and execute a functor based on the conditional compile-time argument.
+         * The functor implementations must have valid C++ syntax but is only evaluated if executed.
+         *
+         * @tparam T_condition execute trueFunctor if the condition is true, else no operation is performed
+         * @tparam T_ConditionTrueFunctor functor type, the functor is only evaluated if T_condition is true
+         * @tparam T_Args types of the arguments for trueFunctor call
```

Comment:
just a technical question:

Is it unlikely that the `falseFunctor` will have possible arguments?
Or should there maybe be `T_ArgsOnTrue` and `T_ArgsOnFalse` for added generality?

Reply:
The arguments should be the same for the false and true  functor.
You can use the lambda functor capture to pass different arguments into the functor.